File: lessons/compiled_lang/java/src/e60.branch.java
import java.io.IOException;

class E60Branch {
    public static void main(String[] args) throws IOException {
        System.out.print("Input a character: ");
        char keypress = (char) System.in.read();
        String s = String.valueOf(keypress);

        if (s.matches("[A-Z]"))
            System.out.println("Uppercase letter");
        else if (s.matches("[a-z]"))
            System.out.println("Lowercase letter");
        else if (s.matches("[0-9]"))
            System.out.println("Digit");
        else
            System.out.println("Punctuation, whitespace, or other");
    }
}

File: lessons/compiled_lang/rust/src/e60.branch.rs
use std::io::{self, Read, Write};

fn main() {
    print!("Input a character: ");
    io::stdout().flush().unwrap();

    let mut buf = [0u8; 1];
    io::stdin().read_exact(&mut buf).unwrap();
    let keypress = buf[0] as char;

    // the standard library has no regex (that's an external crate, and
    //  these lessons stay dependency-free) - matches!() with a char
    //  range pattern is the native stand-in for a single pattern test
    if matches!(keypress, 'A'..='Z') {
        println!("Uppercase letter");
    } else if matches!(keypress, 'a'..='z') {
        println!("Lowercase letter");
    } else if matches!(keypress, '0'..='9') {
        println!("Digit");
    } else {
        println!("Punctuation, whitespace, or other");
    }
}
